nicer event titles

king-11 opened the pull request:
nicer event titles display a few company names else show count
﻿regex split use to remove access token as is not necessary for summary

king-11 commented:
@mal-tee review please closes issue #422

mal-tee commented:
@baltpeter Can you review this please? :D

baltpeter commented on src/my-requests.js:
```diff
@@ -349,6 +349,14 @@ class RequestList extends preact.Component {
+                          .map((item) => item.split(re)[0].trim())
+                          .slice(0, 2)
+                          .join(', ')}`
+                    : `for ${items.length} companies`;
```
Instead of just showing the number here, it would be great to still list first one or two company names and then say "and <n> more companies" or something like that.

king-11 replied:
okay done

baltpeter reviewed (changes requested):
Thanks for the PR! I have left some comments below.

king-11 commented on src/my-requests.js:
```diff
@@ -349,6 +349,14 @@ class RequestList extends preact.Component {
                     } – ${request.reference})`
                 );
             });
+            const re = /\(.+\)$/;
```
okay resolved

king-11 commented:
@baltpeter  rest i have resolved this translation part am not getting a bit; the current changes are as follows 
```js
grouped_requests[group].forEach((id) => {
                let request = this.state.requests[id];
                titles.push(request.recipient.split('\n', 1)[0])
                items.push(
                    `* ${request.recipient.split('\n', 1)[0]} (${
                        request.type === 'custom' && request.response_type
                            ? t(request.response_type, 'generator')
                            : t(request.type, 'my-requests')
                    } – ${request.reference})`
                );
            });
            titles = titles.map((title) => title.length > 10 ? `${title.slice(0,10)}..` : title)
            let heading = ` for ${titles.slice(0,2).join(" ,")}`
            heading += titles.length > 2 ? ` and ${titles.length - 2} more` : ""
```

how can I use t function for this I am bit not sure could you help out a bit

baltpeter commented:
The translation files are in [this directory](https://github.com/datenanfragen/website/tree/master/src/i18n). For example, the English translations for the 'My requests' feature are here:

https://github.com/datenanfragen/website/blob/ce425600cca7111ba1bf187b165462f6536a3352/src/i18n/en.json#L229-L258

You can simply add new translations to that object and then use them like this: `t(your_translation_key, 'my-requests')`

king-11 commented:
@baltpeter so there are three words am using 'for' 'more' and 'and' so i have to add them into the file.

baltpeter reviewed (changes requested):
Thanks for making the changes, this is almost done now. You don't need to add the translations for the other languages (that will be done by our contributors on Weblate) but currently, even the English translations aren't displaying correctly. This is because you also need to pass the context to the `t()` function like this: `t('<translation_key>', 'my-requests')`

king-11 commented:
> Thanks for making the changes, this is almost done now. You don't need to add the translations for the other languages (that will be done by our contributors on Weblate) but currently, even the English translations aren't displaying correctly. This is because you also need to pass the context to the `t()` function like this: `t('<translation_key>', 'my-requests')`

ohh missed that will get in done in just 5 mins

king-11 commented:
@baltpeter i guess finally its done :smile:

baltpeter reviewed (approved):
Thanks again.

king-11 commented:
@baltpeter @mal-tee thanks for helping out throughout 😄 felt great to make contribution to this really amazing project.

mal-tee commented:
In case you haven't seen it: We also have our own little hacktoberfest-thingy: https://www.datarequests.org/blog/hacktoberfest-2020/
